flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -116,15 +119,48 @@ public void registerScalarFunction(String name, ScalarFunction function) {
 	}
 
 	public String[] getUserDefinedFunctions() {
-		return userFunctions.values().stream()
-			.map(FunctionDefinition::toString)
-			.toArray(String[]::new);
+		List<String> result = new ArrayList<>();
+
+		// Get functions in catalog
+		Catalog catalog = catalogManager.getCatalog(catalogManager.getCurrentCatalog()).get();
+		try {
+			result.addAll(catalog.listFunctions(catalogManager.getCurrentDatabase()));
+		} catch (DatabaseNotExistException e) {
+			// Ignore since there will always be a current database of the current catalog
+		}
+
+		// Get built-in functions
+		result.addAll(
+			BuiltInFunctionDefinitions.getDefinitions()
+				.stream()
+				.map(f -> f.getName())
+				.collect(Collectors.toList()));
+
```

Comment:
minor:
Since this method also returns the built-in functions, we may change the method name to another like `getDefinedFunctions` or other?

Reply:
I agree. That can be on its own PR